### PR TITLE
added tcp_user_timeout

### PR DESCRIPTION
**Pull request by Jitendra-Rajak**
#### Describe Bug or Feature
pbs_comm not marking node down when there is job with substate 41.

#### Describe Your Change
Added TCP_USER_TIMEOUT switch alongside the other keepalive switches we have. Currently I have added user_timeout of 50000ms i.e 50 seconds. Increasing the user timeouts allows a TCP connection to survive extended periods without end-to-end connectivity. Decreasing the user timeouts allows applications to "fail fast" if so desired. Otherwise it may take up to 20 minutes with the current system defaults in a normal environment.

#### Attach Test and Valgrind Logs/Output

[logs_50sec_timeout.txt](https://github.com/PBSPro/pbspro/files/3493226/logs_50sec_timeout.txt)

### CODE REVIEW

**Comment by nithinj** [comment]
@Jitendra-Rajak ,
Can you try changing the value in the spec file and verify whether its getting reflected in the logs?

File: src/lib/Libtpp/tpp_dis.c [hunk 1]
```diff
@@ -893,6 +894,7 @@ set_tpp_config(struct pbs_config *pbs_conf,
 	tpp_conf->tcp_keep_idle = DEFAULT_TCP_KEEPALIVE_TIME;
 	tpp_conf->tcp_keep_intvl = DEFAULT_TCP_KEEPALIVE_INTVL;
 	tpp_conf->tcp_keep_probes = DEFAULT_TCP_KEEPALIVE_PROBES;
+	tpp_conf->tcp_user_timeout = DEFAULT_TCP_USER_TIMEOUT;
 
 	/* if set, read them from environment variable PBS_TCP_KEEPALIVE */
 	if ((s = getenv(PBS_TCP_KEEPALIVE))) {
```
Comment (vstumpf) [inline src/lib/Libtpp/tpp_dis.c]: Can you change the comment on line 903 to add tcp_user_timeout to the format.

**Comment by Jitendra-Rajak** [comment]
Hey @nithinj, I have tried changing the env variable values and yes it is reflecting in the logs.

**Comment by Jitendra-Rajak** [comment]
Hey @vstumpf, I have addressed your comments please take the PR for review once again and if there is any further comments do let me know.

Thanks,
Jitendra

**Pull request merged**